zerver/views/registration.py

```diff
@@ -380,7 +387,9 @@ def accounts_register(request: HttpRequest) -> HttpResponse:
                     redirect_url = add_query_to_redirect_url(view_url, query)
                     return HttpResponseRedirect(redirect_url)
             elif not realm_creation:
-                # Since we'll have created a user, we now just log them in.
+                # Since we'll have created a user, we now just update their
+                # default language and log them in.
+                do_change_user_default_language(user_profile, default_language)
```

Comment:
hmm. I am curious why we are not setting the timezone here as well. I only implemented that feature I think,but can't seem to recall why I never set the timezone here. Any idea?

Reply:
I think this piece of code will run for user creation ... and later we are setting the timezone for user_display_setting.. maybe that is why we are not setting the timezone here...